cupyx/scipy/ndimage/filters.py

```diff
@@ -0,0 +1,200 @@
+
+    .. seealso:: :func:`scipy.ndimage.correlate`
+    """
+    return _correlate_or_convolve(input, weights, output, mode, cval, origin,
+                                  False)
+
+
+def convolve(input, weights, output=None, mode='reflect', cval=0.0, origin=0):
+    """Multi-dimensional convolution.
+
+    The array is convolved with the given kernel.
+
+    Args:
+        input (cupy.ndarray): The input array.
+        weights (cupy.ndarray): Array of weights, same number of dimensions as
+            input
+        output (cupy.ndarray, dtype or None): The array in which to place the
+            output.
+        mode (str): The array borders are handled according to the given mode
+            (``'reflect'``, ``'constant'``, ``'nearest'``, ``'mirror'``,
+            ``'wrap'``). Default is ``'reflect'``.
+        cval (scalar): Value to fill past edges of input if mode is
+            ``constant``. Default is ``0.0``.
+        origin (scalar or tuple of scalar): The origin parameter controls the
+            placement of the filter, relative to the center of the current
+            element of the input. Default of 0 is equivalent to
+            ``(0,)*input.ndim``.
+
+    Returns:
+        cupy.ndarray: The result of convolution.
+
+    .. seealso:: :func:`scipy.ndimage.convolve`
+    """
+    return _correlate_or_convolve(input, weights, output, mode, cval, origin,
+                                  True)
+
+
+def _get_output(output, input, shape=None):
+    if shape is None:
+        shape = input.shape
+    if isinstance(output, cupy.ndarray):
+        if output.shape != tuple(shape):
+            raise ValueError('output shape is not correct')
+    else:
+        dtype = output
+        if dtype is None:
+            dtype = input.dtype
+        output = cupy.zeros(shape, dtype)
+    return output
+
+
+def _correlate_or_convolve(input, weights, output, mode, cval, origin,
+                           convolution):
+    if input.dtype in (numpy.complex64, numpy.complex128, numpy.complex256):
+        raise TypeError('Complex type not supported.')
+    if not hasattr(origin, '__getitem__'):
+        origin = [origin, ] * input.ndim
+    else:
+        origin = list(origin)
+    wshape = [ii for ii in weights.shape if ii > 0]
+    if len(wshape) != input.ndim:
+        raise RuntimeError('filter weights array has incorrect shape.')
+    if convolution:
+        weights = weights[tuple([slice(None, None, -1)] * weights.ndim)]
+        for ii in range(len(origin)):
+            origin[ii] = -origin[ii]
+            if weights.shape[ii] % 2 == 0:
+                origin[ii] -= 1
+    for _origin, lenw in zip(origin, wshape):
+        if (lenw // 2 + _origin < 0) or (lenw // 2 + _origin > lenw):
+            raise ValueError('invalid origin')
+    if mode not in ('reflect', 'constant', 'nearest', 'mirror', 'wrap'):
+        msg = 'boundary mode not supported (actual: {}).'.format(mode)
+        raise RuntimeError(msg)
```

Comment:
Have raised issue ( https://github.com/cupy/cupy/issues/2525 ) to track adding support for `mode`.